### ISSUE TITLE
Creating a base representer

### ISSUE THREAD

**Issue by tastycake** [trailblazer/representable#216]
I'm having trouble understanding why my tests are failing when trying to DRY up my representers.

```ruby
module MyModule::BaseRepresenter
  include Representable::JSON
  
  property :id
  property :name
  property :created_at
  property :updated_at
end

module MyModule::CompanyRepresenter
  include MyModule::BaseRepresenter

  property :logo
  property :category
end
```
When I run my tests, they fail with ``NoMethodError: undefined method 'register_feature'``

If I include ``Representable::JSON`` in ``CompanyRepresenter`` everything works fine. The objects return have the properties from both ``Base`` and ``Company``.

I may be misunderstanding something about mixins, however I would assume that ``Representable::JSON`` would propagate to ``CompanyRepresenter``.

**Comment by apotonick** [comment]
No, you need to include `Representable::JSON` in every "end user" representer.

**Comment by apotonick** [comment]
I *think* that was due to the idea that we can have generic representers (e.g. for JSON and XML) or something like that. But I agree with your idea. I also remember that we tried to implement it and it ended up being super complicated... but that might've been due to the lack of understanding back then.

Anyway, in "vanilla" Ruby, including a module `A` into `B` doesn't automatically re-include `A` into `C` when including `B` into `C` - which is very important for gems like Representable and their use of the `included` hook. Is that where your confusion is coming from?
